### PR TITLE
python3Packages.desktop-entry-lib: init at 3.1

### PR DESCRIPTION
**Pull request by Madouura**
## Description of changes
For ricing purposes

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/desktop-entry-lib/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+, fetchFromGitea
+, gitUpdater
+, setuptools
+}:
+
+buildPythonPackage rec {
+  pname = "desktop-entry-lib";
+  version = "3.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.9";
```
Comment (fabaff) [inline pkgs/development/python-modules/desktop-entry-lib/default.nix:14]: This was changed recently and it the new way to specify the "format".

```suggestion
  pyproject = true;

  disabled = pythonOlder "3.9";
```

File: pkgs/development/python-modules/desktop-entry-lib/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  # We could use fetchPypi, but then the tests won't run
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "JakobDev";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-+c+FuLv88wc4yVw3iyFFtfbocnWzTCIe2DS0SWoj+VI=";
```
Comment (fabaff) [inline pkgs/development/python-modules/desktop-entry-lib/default.nix:22]: It's a SRI hash.

```suggestion
    hash = "sha256-+c+FuLv88wc4yVw3iyFFtfbocnWzTCIe2DS0SWoj+VI=";
```

File: pkgs/development/python-modules/desktop-entry-lib/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  nativeBuildInputs = [
+    pytestCheckHook
+    setuptools
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/desktop-entry-lib/default.nix:28]:
```suggestion
  nativeBuildInputs = [
    setuptools
  ];

  nativeCheckInputs = [
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/desktop-entry-lib/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  pythonImportsCheck = [ "desktop_entry_lib" ];
+  passthru.updateScript = gitUpdater { };
```
Comment (fabaff) [inline pkgs/development/python-modules/desktop-entry-lib/default.nix:31]: No needed Python module usually works fine with `nix-update`.

```suggestion
```

File: pkgs/development/python-modules/desktop-entry-lib/default.nix (hunk 2 above)
Comment (Madouura) [reply pkgs/development/python-modules/desktop-entry-lib/default.nix:22]: Oh my, surprised I didn't catch that.

**Pull request merged**